sdk/storage/Azure.Storage.Files.DataLake/src/DataLakeUriBuilder.cs

```diff
@@ -241,6 +242,54 @@ public Uri ToUri()
             return _uri;
         }
 
+        /// <summary>
+        /// Gets the blob Uri.
+        /// </summary>
+        internal Uri ToBlobUri()
+        {
+            // AccountName cannot be null
+            // Host.Length mst be at least 5 character longth than AccountName.Length to account for ".dfs."
+            // Host must begin with "[AccountName]."
+            // The substring of Host immediately following "[AccountName]." must be "dfs."
+            if (AccountName != null
+                && Host.Length >= AccountName.Length + 5
+                && Host.Substring(0, AccountName.Length + 1).Equals(AccountName + ".", StringComparison.OrdinalIgnoreCase)
+                && Host.Substring(AccountName.Length + 1, 4).Equals(Constants.DataLake.DfsUriSuffix + ".", StringComparison.OrdinalIgnoreCase))
+            {
+                StringBuilder stringBuilder = new StringBuilder(Host);
+
+                // Replace "dfs" with "blob"
+                stringBuilder.Replace(Constants.DataLake.DfsUriSuffix, Constants.DataLake.BlobUriSuffix, AccountName.Length + 1, 3);
+                Host = stringBuilder.ToString();
+            }
+
+            return ToUri();
+        }
+
+        /// <summary>
+        /// Gets the dfs Uri.
+        /// </summary>
+        internal Uri ToDfsUri()
+        {
+            // AccontName cannot be null
+            // Host.Length must be at least 6 characters longther than AccountName.Length to account for ".blob."
+            // Host must begin with "[AccountName]."
+            // The substring of Host immediately following "[AccountName]." must be ".blob."
+            if (AccountName != null
+                && Host.Length >= AccountName.Length + 6
+                && Host.Substring(0, AccountName.Length + 1).Equals(AccountName + ".", StringComparison.OrdinalIgnoreCase)
+                && Host.Substring(AccountName.Length + 1, 5).Equals(Constants.DataLake.BlobUriSuffix + ".", StringComparison.OrdinalIgnoreCase))
+            {
+                StringBuilder stringBuilder = new StringBuilder(Host);
+
+                // Replace "blob" with "dfs"
+                stringBuilder.Replace(Constants.DataLake.BlobUriSuffix, Constants.DataLake.DfsUriSuffix, AccountName.Length + 1, 4);
+                Host = stringBuilder.ToString();
+            }
+
+            return ToUri();
+        }
+
```

Comment:
Any chance we can get rid of the duplicate GetAccountNameFromDomain here in favor of the extension method in UriExtension

Reply:
We could, I'm just following the comments from Ted in the original Data Lake PR, which are copied to this Issue: https://github.com/Azure/azure-sdk-for-net/issues/8386

Reply:
This comment is about GetAccountNameFromDomain. I couldn't add it directly to the line because you haven't changed that code.

Reply:
I'd prefer we address it with a GitHub issue.

Reply:
370

Reply:
Removed.